[RHACS] ROX-18793: openshift monitoring (#63050)
openshift/openshift-docs: configuration/monitor-acs.adoc

[id="monitor-acs"]
= Monitoring {product-title-short}
include::modules/common-attributes.adoc[]
:context: monitor-acs

toc::[]

== Monitoring with {osp}

[role="_abstract"]
link:https://docs.openshift.com/container-platform/4.13/monitoring/monitoring-overview.html[{ocp}] includes a preconfigured,
preinstalled, and self-updating monitoring stack that provides monitoring for core platform components.

> {product-title-short} has support for {osp} monitoring by using an encrypted and authenticated endpoint accessible only by the Prometheus Operator.

Reviewer: ```suggestion
{product-title-short} exposes metrics to {osp} monitoring via an encrypted and authenticated endpoint.
```

While OpenShift monitoring uses Prometheus operator under the hood, it is an implementation detail. Also it's the Prometheus pod, not the Prometheus operator pod that scrapes the metrics.

[IMPORTANT]
====

> If you have previously configured monitoring with the Prometheus Operator,
> consider disabling it and use OpenShift Monitoring instead that is enabled by default on {osp} 4.
> Having multiple ServiceMonitors could result in duplicating data.

Reviewer: ```suggestion
If you have previously configured monitoring with the Prometheus Operator,
consider removing your custom ServiceMonitor. {product-title-short} ships with a pre-configured ServiceMonitor for {osp} monitoring.

Multiple ServiceMonitors may result in duplicated scraping.
```

====

include::modules/enable-openshift-monitoring-central-operator.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources
* xref:../installing/installing_ocp/install-central-config-options-ocp.adoc#install-central-config-options-ocp[Central configuration options using the Operator]

include::modules/enable-openshift-monitoring-central-helm.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources
* xref:../installing/installing_other/install-central-other.adoc#change-config-options-after-deployment-central-services_install-central-other[Changing configuration options after deploying the central-services Helm chart]

== Monitoring with custom Prometheus

[role="_abstract"]
link:https://prometheus.io/[Prometheus] is an open-source monitoring and alerting platform.
You can use it to monitor health and availability of Central and Sensor components of {rh-rhacs-first}. When you enable monitoring, {product-title-short} creates a new monitoring service on port number 9090 and a network policy allowing inbound connections to that port.

[NOTE]
====

> This monitoring service is exposes an endpoint that is not encrypted by TLS and has no authorization.

Reviewer: ```suggestion
This monitoring service exposes an endpoint that is not encrypted by TLS and has no authorization.
```

Use this only when you do not want to use {osp} monitoring.
====

> include::modules/enable-monitoring-central-operator.adoc[leveloffset=+1]

Reviewer 2: This is actually still not resolved, sorry. This looks like an old version of the enable-openshift-monitoring-central-operator.adoc file, which is properly called on line 22. This file included here hasn't been updated and doesn't contain the latest changes. I think this include line should be deleted, and the file enable-monitoring-central-operator.adoc should be deleted.

Author: Ah, good catch. I added more context to it but I'm not sure if it's better now

![image](https://github.com/openshift/openshift-docs/assets/1616386/bb4f1ba8-9bc9-4d76-81f5-0ddc9d3cf6d0)

Author: maybe we should change the "old" names and add "custom" there ?

Reviewer: I feel like we now have way too many sections. Seems very complicated for a feature that is essentially working out of the box without any interaction needed. Maybe make the `... by using` sub-sections?

[id="monitor-acs"]
= Monitoring {product-title-short}
include::modules/common-attributes.adoc[]
:context: monitor-acs

toc::[]

== Monitoring with {osp}

[role="_abstract"]
link:https://docs.openshift.com/container-platform/4.13/monitoring/monitoring-overview.html[{ocp}] includes a preconfigured,
preinstalled, and self-updating monitoring stack that provides monitoring for core platform components.
{product-title-short} exposes metrics to {osp} monitoring via an encrypted and authenticated endpoint.

[IMPORTANT]
====
If you have previously configured monitoring with the Prometheus Operator,
consider removing your custom ServiceMonitor resources. {product-title-short} ships with a pre-configured ServiceMonitor for {osp} monitoring.

> 

Reviewer 2: remove blank line & keep next sentence with rest of admonition

[id="monitor-acs"]

> = Monitoring ACS

Reviewer 2: ```suggestion
= Monitoring {product-title-short}
```

include::modules/common-attributes.adoc[]
:context: monitor-acs

toc::[]

> = Monitoring with OpenShift 4

Reviewer 2: ```suggestion
= Monitoring with {osp} 4
```

> = Monitoring with OpenShift 4

Reviewer: Isn't this redundant, given that we render the docs per OpenShift version?

[role="_abstract"]

> link:https://docs.openshift.com/container-platform/4.13/monitoring/monitoring-overview.html[OpenShift Container Platform] includes a preconfigured,

Reviewer 2: ```suggestion
link:https://docs.openshift.com/container-platform/4.13/monitoring/monitoring-overview.html[{ocp}] includes a preconfigured,
```

preinstalled, and self-updating monitoring stack that provides monitoring for core platform components.

> {product-title-short} has support for OpenShift Monitoring via encrypted and authenticated endpoint accessible only by Prometheus Operator.

Reviewer 2: ```suggestion
{product-title-short} has support for {osp} monitoring by using encrypted and authenticated endpoint accessible only by the Prometheus Operator.
```

[IMPORTANT]
====

> If you have previously configured monitoring with Prometheus Operator you may consider disabling it as it will be enabled by default on OpenShift 4.

Reviewer 2: ```suggestion
If you have previously configured monitoring with the Prometheus Operator, consider disabling it, because it is enabled by default on {osp} 4.
```

> Having multiple ServiceMonitors could result in duplicating data and wrong query results.

Reviewer: Shouldn't we rather advice to use the OpenShfit monitoring? I'm also not sure about the warning about wrong query results, it sounds a bit too scary for what it is imo.

====

include::modules/enable-openshift-monitoring-central-operator.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources
* xref:../installing/installing_ocp/install-central-config-options-ocp.adoc#install-central-config-options-ocp[Central configuration options using the Operator]

include::modules/enable-openshift-monitoring-central-helm.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources
* xref:../installing/installing_other/install-central-other.adoc#change-config-options-after-deployment-central-services_install-central-other[Changing configuration options after deploying the central-services Helm chart]

= Monitoring with Prometheus

[role="_abstract"]
link:https://prometheus.io/[Prometheus] is an open-source monitoring and alerting platform.
You can use it to monitor health and availability of Central and Sensor components of {rh-rhacs-first}. When you enable monitoring, {product-title-short} creates a new monitoring service on port number `9090` and a network policy allowing inbound connections to that port.

[NOTE]
====

> Monitoring service is unencrypted and has no authorization. It should be used only when you do not want to use OpenShift Monitoring.

Reviewer 2: ```suggestion
This monitoring service is unencrypted and has no authorization. Use this only when you do not want to use {osp} monitoring.
```

Reviewer: Maybe a bit pedantic, but what does the "service is unencrypted" mean? Isn't the service resource always a plaintext manifest? I think it's better to be more precise here and state that the service exposes an endpoint that is not encrypted by TLS.
